### PR TITLE
Add `CNOT` support

### PR DESCRIPTION
**Pull request by rmoyard**
This pull request adds the support of CNOT gates by using a decomposition with the MS gate. It is directly related to the issue https://github.com/PennyLaneAI/pennylane-aqt/issues/14.

### CODE REVIEW

File: tests/test_device.py [hunk 1]
```diff
@@ -193,6 +193,24 @@ def test_apply_operation_hadamard(self, wires):
 
         assert dev.circuit == [["X", 1.0, wires], ["Y", -0.5, wires]]
 
+    @pytest.mark.parametrize("wires", [[0, 1], [1, 0], [1, 2], [2, 1]])
+    def test_operation_cnot(self, wires):
```
Comment (antalszava) [inline tests/test_device.py:197]: Maybe it's worth also-double checking that the decomposition produces the correct matrix representation for CNOT.

**Review by antalszava — COMMENTED** [review]
Hi @rmoyard, thank you for tackling this so quick! :partying_face: :heart_eyes: Overall this looks good, just had 1-2 comments. One thing I'm wondering about is if we'd need to consider leakage for the noisy simulator (and if the `MS` gate would not be applied when applying a `CNOT` on the noisy device).

File: tests/test_device.py (hunk 1 above)
Comment (rmoyard) [reply tests/test_device.py:197]: I think there is no matrix representation in `Aqt`, but I believe the decomposition is correct (found in multiple papers).

File: tests/test_device.py [hunk 2]
```diff
@@ -193,6 +193,24 @@ def test_apply_operation_hadamard(self, wires):
...
+    @pytest.mark.parametrize("wires", [[0, 1], [1, 0], [1, 2], [2, 1], [0,2], [2,0]])
+    def test_operation_cnot(self, wires):
+        """Tests that the _apply_operation method correctly populates the circuit
+        queue when a PennyLane CNOT operation is provided."""
+
+        dev = AQTDevice(3, api_key=SOME_API_KEY)
+        assert dev.circuit == []
+
+        dev._apply_operation(qml.CNOT(wires=wires))
+
```
Comment (antalszava) [inline tests/test_device.py:205]:
```suggestion

    # Note: the original parameters used in PennyLane are divided by pi as per AQT convetion
```

**Review by antalszava — APPROVED** [review]
Looks good to me! :100: Thank you for the fast and great work :tada: :slightly_smiling_face:

**Comment by antalszava** [comment]
[ch8530]

**Pull request merged**